wgpu-core/src/device/resource.rs

```diff
@@ -1832,7 +1832,16 @@ impl<A: HalApi> Device<A> {
                 }
                 (size.get(), end)
             }
-            None => (buffer.size - bb.offset, buffer.size),
+            None => {
+                if buffer.size < bb.offset {
+                    return Err(Error::BindingRangeTooLarge {
+                        buffer: bb.buffer_id,
+                        range: bb.offset..bb.offset,
```

Comment:
My only concern is that this faux range may make the error message very confusing for the user, as this range never actually shows up in their code.